authentication/http.go

```diff
@@ -54,6 +54,30 @@ func WithTenantID(tenantIDs map[string]string) Middleware {
 	}
 }
 
+// EnforceAccessTokenPresent enforces that the Authorization header is present in the incoming request.
+// Otherwise, it returns an error.
+func EnforceAccessTokenPresent(urlPath string) Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if !strings.HasSuffix(r.URL.Path, urlPath) {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			rawToken := r.Header.Get("Authorization")
+			if rawToken == "" {
+				http.Error(w, "couldn't find the authorization header", http.StatusBadRequest)
+				return
+			}
+			if !strings.Contains(rawToken, "Bearer ") {
+				http.Error(w, "malformed authorization header", http.StatusBadRequest)
+				return
+			}
```

Comment:
I tend to agree with @matej-g here, but just out of interest, does the presence of an `Authorization` header alone avoid the server sending the client into this redirect hotloop or does the situation also occur wit an invalid/expired token?

Reply:
We could leave it out, at the cost of decreasing protection from tenants "DoSing" the identity provider with bad configuration. 

The idea of doing this very simple check here is exactly to hard-stop a request **before** it reaches the OIDC middleware, because this kind of failure there means a redirect to the identity provider's login page.

What do you think?

Reply:
what about situations other than OIDC, I see support for mTLS and OpenShift authenticator for example? 🤔

Reply:
I don't think Prometheus remote write supports OpenShift authenticator, so this discards the OpenShift authenticator at least, correct?

I am investigating how can we make this not break the mTLS authentication. Sadly there aren't tests for this. :(

Reply:
> what about situations other than OIDC, I see support for mTLS and OpenShift authenticator for example? thinking

That's a good point actually, should this be then part of the OIDC middleware rather, as it applies only to tenants with OIDC authentication providers?

Reply:
Thanks for the comments too, @PhilipGough and @esnible. I am taking your suggestions into account. 👍